### PR TITLE
Storage: simplify retrieving the file names from the response frame

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
This PR aims to simplify retrieving the file names from the response frame. Adding a wrapper over the response frame will make it easier for us to change the response DTO format in the future, as all the required refactoring will be localized.

### CODE REVIEW

File: pkg/services/store/types.go [hunk 1]
```diff
@@ -95,3 +95,40 @@ type RootStorageMeta struct {
...
+	nameListFrameField        listFrameField = "name"
+	descriptionListFrameField listFrameField = "description"
+	mediaTypeListFrameField   listFrameField = "mediaType"
+	storageTypeListFrameField listFrameField = "storageType"
+	readOnlyListFrameField    listFrameField = "readOnly"
+	builtInListFrameField     listFrameField = "builtIn"
+	sizeListFrameField        listFrameField = "size"
+)
+
+func (s *StorageListFrame) GetFileNames() []string {
```
Comment (ArturWierzbicki) [inline pkg/services/store/types.go:116]: I tried to come up with something abstract, but, due to the lack of generics & mapped types, all other approaches resulted in more hassle for the client (casting) - not sure if there is any better alternative

File: pkg/services/store/types.go [hunk 2]
```diff
@@ -95,3 +95,40 @@ type RootStorageMeta struct {
 
 	Config RootStorageConfig `json:"config"`
 }
+
+type StorageListFrame struct {
+	*data.Frame
+}
+
+type listFrameField string
```
Comment (sakjur) [inline pkg/services/store/types.go:103]: AFAICT, we're casting every one of these back to string on every use, what does this type give us?

**Review by sakjur — COMMENTED** [review]
We could merge this as-is, it seem to do what it says reasonably well (based on a read-thru). I have some suggestions for general readability improvements, just take a pass over them and lmk what you think and I'll approve after if nothing comes up 🙂

File: pkg/services/store/types_test.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+import (
+	"testing"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	exampleListFrameJSON = `{
```
[the remark this answers is not in the document]
Comment (ArturWierzbicki) [reply pkg/services/store/types_test.go]: neat, changed!

File: pkg/services/store/types.go [hunk 4]
```diff
@@ -95,3 +95,40 @@ type RootStorageMeta struct {
 
 	Config RootStorageConfig `json:"config"`
 }
+
+type StorageListFrame struct {
+	*data.Frame
```
[the remark this answers is not in the document]
Comment (ArturWierzbicki) [reply pkg/services/store/types.go:100]: This really is just a type alias - ideally I would be able to just say `type StorageListFrame *data.Frame` but I failed to make the receiver function work with such declaration

The practical implication of having `Frame` as a field is the hassle of removing it when serializing the frame to JSON

**Comment by ryantxu** [comment]
@kylebrandt -- here is an interesting example where we want to use `DataFrame` (since it has a flexible structure and integrates with panels well), but need to programmatically access things in a more friendly way.

This seems like a pattern worth exploring.  I am not wild about this much ceremony for a simple "get field names as string" command

**Review by sakjur — APPROVED** [review]

**Pull request merged**